src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -2306,8 +2311,19 @@ int fmi3_xml_handle_ModelVariables(fmi3_xml_parser_context_t *context, const cha
         }
         jm_vector_qsort(jm_voidp)(&md->variablesByVR, fmi3_xml_compare_vr_and_original_index);
 
-        // Error check duplicate VRs:
         if (nVars > 0) {  // nVars=0 would cause integer overflow in the loop condition
+            /* Error check duplicate variable names */
+            for (size_t i = 0; i < nVars-1; ++i) {
+                const char* name1 = jm_vector_get_item(jm_named_ptr)(&md->variablesByName, i).name;
+                const char* name2 = jm_vector_get_item(jm_named_ptr)(&md->variablesByName, i+1).name;
+                if(strcmp(name1, name2) == 0) {
```

Comment:
This looks like we expect the list to be sorted. Is that the case?

Reply:
Yes, there is a `jm_vector_qsort(jm_named_ptr)(&md->variablesByName, jm_compare_named);` a few lines above, which sorts the list.

Reply:
ok